test/radfiletests.cpp

```diff
@@ -139,42 +139,91 @@ TEST(RadfileTests, SingleLine)
-      break;
+    stadic::RadFileData radData;
+    ASSERT_TRUE(radData.addRad("materialsingleline.rad"));
+    ASSERT_EQ(6, radData.primitives().size());
+    stadic::RadPrimitive *black = new stadic::PlasticMaterial(0, 0, 0, 0, 0);
+    black->setModifier(stadic::RadPrimitive::sharedVoid());
+    black->setName("black");
+    radData.addPrimitive(black);
+    ASSERT_EQ(7, radData.primitives().size());
+    std::shared_ptr<stadic::RadPrimitive> found = nullptr;
+    for (auto poly : radData.geometry()) {
+        if (poly->name() == "l_wall") {
+            found = poly;
+            break;
+        }
     }
-  }
-  ASSERT_EQ(nullptr, found);
-  for (auto matl : radData.materials()) {
-    if (matl->name() == "l_wall") {
-      found = matl;
-      break;
+    ASSERT_EQ(nullptr, found);
+    std::shared_ptr<stadic::RadPrimitive> foundf = radData.findGeometry([](std::shared_ptr<stadic::RadPrimitive> p){return p->name() == "l_wall"; });
+    ASSERT_EQ(nullptr, foundf);
+    for (auto matl : radData.materials()) {
+        if (matl->name() == "l_wall") {
+            found = matl;
+            break;
+        }
     }
-  }
-  ASSERT_NE(nullptr, found);
-  found = nullptr;
-  for (auto matl : radData.materials()) {
-    if (matl->name() == "black") {
-      found = matl;
-      break;
+    ASSERT_NE(nullptr, found);
+    foundf = radData.findPrimitive([](std::shared_ptr<stadic::RadPrimitive> p){return p->name() == "l_wall"; });
+    ASSERT_EQ(found, foundf);
+    found = nullptr;
+    for (auto matl : radData.materials()) {
+        if (matl->name() == "black") {
+            found = matl;
+            break;
+        }
     }
-  }
-  ASSERT_NE(nullptr, found);
+    ASSERT_NE(nullptr, found);
+    foundf = radData.findMaterial([](std::shared_ptr<stadic::RadPrimitive> p){return p->name() == "black"; });
+    ASSERT_EQ(found, foundf);
 }
 
 TEST(RadfileTests, Aliasing)
 {
-  ASSERT_TRUE(false);
+    // Set up the first set of data
+    stadic::RadFileData radData;
+    ASSERT_TRUE(radData.addRad("materialsingleline.rad"));
+    ASSERT_EQ(6, radData.primitives().size());
+    stadic::RadPrimitive *black = new stadic::PlasticMaterial(0, 0, 0, 0, 0);
+    black->setModifier(stadic::RadPrimitive::sharedVoid());
+    black->setName("black");
+    std::shared_ptr<stadic::RadPrimitive> blackMaterial = radData.addPrimitive(black);
+    ASSERT_NE(nullptr, blackMaterial);
+    ASSERT_EQ(7, radData.primitives().size());
+    ASSERT_TRUE(radData.addRad("simple.rad"));
+    ASSERT_EQ(43, radData.primitives().size());
+    std::shared_ptr<stadic::RadPrimitive> windowMaterial;
+    windowMaterial = radData.findPrimitive([](std::shared_ptr<stadic::RadPrimitive> p){return p->name() == "l_window"; });
+    ASSERT_NE(nullptr, windowMaterial);
+    ASSERT_TRUE(radData.setAlias(blackMaterial, windowMaterial));
+    ASSERT_EQ(1, radData.aliases().size());
+
+    // Copy the first set, then make additional changes
+    stadic::RadFileData otherRad = radData;
```

Comment:
@CraigCasey @RMistrick There might be a better way to point you at this code, but this is easiest right now. This test is mainly about aliasing, but also note that it makes copies of the RadFileData object that gets both the primitives and the aliases. All of those lines in daylight.cpp that are leaking memory can probably be replaced with something like this.